docs/articles/documentation/guides/basic-guides/best-practices.md

```diff
@@ -0,0 +1,333 @@
+
+    await page.selectFeature(2);
+    await page.typeName('Peter');
+    await page.clickSubmit();
+
+});
+```
+
+The page model object handles the identification of items and common operations. Tests are then more readable and less brittle because there is no duplicate code. Notice how this test doesn't require a test controller (`t`) since the page object handles all the actions.
+
+The [Page Model](../../guides/concepts/page-model.md) article covers the use of page objects in greater detail.
+
+A code example is available in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+## Use of Roles for Login
+
+Handle authentication during your tests with [User Roles](../../guides/advanced-guides/authentication.md#user-roles). Roles allow you to wrap authentication logic and credentials in a reusable object.
+
+Define roles in a separate file with the [Role](../../reference/test-api/role/constructor.md) constructor and include them in your tests with the [t.useRole](../../reference/test-api/testcontroller/userole.md) method. The following example makes use of two user roles:
+
+```js
+import { Role } from 'testcafe';
+
+const regularUser = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const admin = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+
+fixture `My Fixture`
+    .page('./my-page.html');
+
+test('My test', async t => {
+    await t
+        .useRole(role);
+});
+```
+
+See the [Authentication](../../guides/advanced-guides/authentication.md) article for more in-depth information.
+
+## File Structure
+
+Follow these guidelines to keep your test structure manageable and "clean":
+
+* Use a page model to store Selectors and compound actions that are used across your app often. For instance, a page model function can contain all steps that are necessary to perform an action.
+
+* Put all the page model files into one directory. If your application is divided logically into components or subsystems, split up the associated page model objects into separate files.
+
+> You can find an example of a page model in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+* Create a `testcaferc.json` file in the root directory of the project. It enables you to fine-tune your tests, which can be useful in CI/CD systems. For more information, read the [Configuration File](../../reference/configuration-file.md) article.
+
+* Define one `fixture` in every test file. While it's technically possible to define multiple, it may lead to confusion as your suite grows.
+
+* A `fixture`s should aggregate tests that relate to each other by meaning or testing scenario.
```

Comment:
> A `fixture` should aggregate related tests. For example, put all your authentication-related tests into one fixture.

Reply:
I see. Can we say "category" instead of "meaning"? For example:

`* A `fixture` should aggregate tests based on category or testing scenario.`

Reply:
I've since reconsidered this sentence. It's now:

> A `fixture` should aggregate related tests. For example, put all your authentication-related tests into one fixture.

Reply:
Thanks for the new suggestion. Can we make it more specific than "put"? For example, does the developer "insert" these tests into one fixture?

Also, "your" is not necessary. We can simply say "all authentication-related tests".

Reply:
If 'place' works, I'd rather use it:

> A fixture should aggregate related tests. For example, place all authentication-related tests into one fixture.

If it doesn't 'insert' is fine (although I feel like it's a tad too specific):

> A fixture should aggregate related tests. For example, insert all authentication-related tests into one fixture.

Reply:
If you feel that 'insert' is too specific, then 'place' works. Please apply your suggestion:

`A fixture should aggregate related tests. For example, place all authentication-related tests into one fixture.`

Reply:
Done

Reply:
Thank you!